tensorboard/components/plugin_util/plugin-guest.ts

```diff
@@ -12,19 +12,27 @@ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 See the License for the specific language governing permissions and
 limitations under the License.
 ==============================================================================*/
-import {IPC, Message, MessageType, PayloadType} from './message.js';
-
-class GuestIPC extends IPC {
-  /**
-   * payload must be JSON serializable.
-   */
-  sendMessage(type: MessageType, payload: PayloadType): Promise<PayloadType> {
-    return this.sendMessageToWindow(window.parent, type, payload);
-  }
+import {IPC, Message} from './message.js';
+
+/**
+ * This code is part of a public bundle provided to plugin authors,
+ * and runs within an IFrame to setup communication with TensorBoard's frame.
+ */
+if (!window.parent) {
+  throw Error(
+    'This library must be run from within a loaded TensorBoard dynamic plugin.'
```

Comment:
Thanks for adding the invariance. `...be run from within` part reads a bit funny to me. Can this be reworded as `The library must run within a TensorBoard iframe-based plugin.`. (dynamic-ness is irrelevant too)

Reply:
Done